### DISCUSSION TITLE
Option to launch Chromium in non-headless (visible) mode

### DISCUSSION THREAD

**Discussion by ghost** [cline/cline#448] [Feature Requests]
It would be super nice to be able to "show" the AI certain pages behind a login in a web app (after navigation by user), so the AI can inspect them (screenshot and logs).

---

Would it be possible to enable launching the chromium instance in non-headless mode? Or do I miss something why that would cause issues. (I don't know much about these chromium tools.)

The parameter could be specified here, I think in `UrlContentFetcher.ts`  (using a configuration parameter perhaps)
```
		this.browser = await stats.puppeteer.launch({
		        headless: true,
```

If opening up chromium with GUI would cause difficulties with the current workflow interaction, maybe there could be a separate tool for the AI to explicitly launch the GUI and ask the user to navigate to the page?

**Comment by jtac** [comment]
Right now i've created an "ai assistant" profile that claude-dev can access by giving a specific API key in the headers, then saving the page using curl, then viewing the saved page but it's very clunky, slow and token intensive.

It would be great if claude-dev could view pages that require a login when creating something like a Django site. This would be one way, i'll create another suggestion to make claude-dev able to actually pass a login.
